### ISSUE TITLE
[Q]How do I get the best parameter config easily after using sweep?

### ISSUE THREAD

**Issue by Jing25** [wandb/client#3244]
This is probably a dumb question. I'm new to wandb and I tried sweep for hyperparamter searching. If I'm interested in a couple of the experiments and hyperparameters in a sweep run, and I want to use the same hyperparameters to train a model and do more tests. How should I do that?

**Comment by ramit-wandb** [comment]
Hi @Jing25,

This should be easily possible through our [Public API](https://docs.wandb.ai/ref/python/public-api). I wrote a small script for you to achieve this:

```python3
api = wandb.Api()
sweep = api.sweep(f"{ENTITY}/{PROJECT}/{SWEEP_ID}")

best_run = sweep.best_run()
print(best_run.config)
```

where `ENTITY`, `PROJECT` and `SWEEP_ID` are strings referring to your entity name, project name and sweep id respectively.

`sweep.best_run` finds the best run based on the metric you specified in your sweep config. If you want to find the best run based on some other metric, you would need to run a loop over `sweep.runs` and compare the runs manually.

Thanks,
Ramit

**Comment by Jing25** [comment]
Awesome! Thank you very much for your clear and detailed answer. It works perfect for me. I will close this issue.

**Comment by nranabhat** [comment]
I'm having a related issue. I'm able to use  `best_loss = sweep.best_run().summary_metrics['loss']` to find the lowest value of loss (float value) in a given sweep. The problem is that I believe `wandb.log()` is not updating wandb fast enough before `sweep.best_run().summary_metrics['loss']` is executed. This leads to a _KeyError('loss')_ exception on the first epoch. I've tried using the `commit=true` parameter but this doesn't do the trick. 

Is there a way to make wandb.log() update wandb before moving onto new code? Here is my code:  

`        for epoch in range(config.epochs):

            avg_loss = train_epoch(network, trainloader, optimizer, scheduler)

            # log loss
            wandb.log({"loss": avg_loss, "epoch": epoch}, commit=True)

            # update best_loss for next call to checkpoint_saver
            best_loss = sweep.best_run().summary_metrics['loss'] 

            # save weights of current best epoch
            if avg_loss <= best_loss:
              checkpoint_saver(network, avg_loss)`

Thanks
